src/arduino.cc/builder/ctags_target_file_saver.go

```diff
@@ -59,3 +67,53 @@ func (s *CTagsTargetFileSaver) Run(context map[string]interface{}) error {
 
 	return nil
 }
+
+func saveLinesContainingDirectivesAndSketch(src string, tofind []string) string {
+	lines := strings.Split(src, "\n")
+
+	saveLine := false
+	minimizedString := ""
+
+	for _, line := range lines {
+		if (saveLine || startsWithHashtag(line)) {
```

Comment:
Why save all the # lines? AFAICS there are only line markers left in the output, and only the ones that are seen while `saveLine` is (or becomes) true are relevant I think?